libraries/src/MVC/Model/FormModel.php

```diff
@@ -98,17 +98,17 @@ public function checkin($pk = null)
 				return false;
 			}
 
-			$checkedOutField = $table->getColumnAlias('checked_out');
-			$checkedOutTimeField = $table->getColumnAlias('checked_out_time');
-
 			// If there is no checked_out or checked_out_time field, just return true.
-			if (!$table->hasField($checkedOutField) || !$table->hasField($checkedOutTimeField))
+			if (!$table->hasField('checked_out') || !$table->hasField('checked_out_time'))
 			{
 				return true;
 			}
 
+			$checkedOutField = $this->getColumnAlias('checked_out');
+			$checkedOutTimeField = $this->getColumnAlias('checked_out_time');
+
 			// Check if this is the user having previously checked out the row.
-			if ($table->{$checkedOutField} > 0 && $table->{$checkedOutField} != $user->get('id') && !$user->authorise('core.admin', 'com_checkin'))
+			if ($table->$checkedOutField > 0 && $table->$checkedOutTimeField != $user->get('id') && !$user->authorise('core.admin', 'com_checkin'))
```

Comment:
```suggestion
			if ($table->$checkedOutField > 0 && $table->$checkedOutField != $user->get('id') && !$user->authorise('core.admin', 'com_checkin'))
```

Do not compare time with user_id.